docs/memory.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Using the Memory View
+---
+
+* toc
+{:toc}
+
+## What is it?
+
+Dynamically allocated Dart objects e.g., ```new ClassName()``` live in a portion of memory called the heap.
```

Comment:
Nit: tag the code snippet as a dart snippet.
It would be ideal to give an example that doesn't use the new keyword as it is optional in Dart now so I don't have a great idea.

Reply:
Allocated Dart objects created using a class constructor see [Dart Using constructors](https://www.dartlang.org/guides/language/language-tour#using-constructors) e.g.,
```dart
new MyClass()
```
or
```dart
MyClass()
```
live in a portion of memory called the heap.